api-gateway/src/main/resources/static/scripts/visits/visits.template.html

```diff
@@ -482,49 +482,100 @@ <h1></h1>
 
         renderCalendar();
     });
-
-    //select dates
-    document.querySelector('.days').addEventListener('click',(event) => {
-
+/*    document.querySelector('.days').addEventListener('click', (event) => {
         let target = event.target.dataset.target;
 
         if(target !== undefined) {
             if (target.substring(0, 13) === "previous-page") {
                 selectedDate = target.substring(14, 16).trim();
-
                 date.setMonth(date.getMonth() - 1);
-
             } else if (target.substring(0, 9) === "next-page") {
                 selectedDate = target.substring(10, 12).trim();
-
                 date.setMonth(date.getMonth() + 1);
-
             } else {
                 selectedDate = target;
                 date.setDate(parseInt(target));
             }
-            selectedTime = "";
-        }
-        if (self) {
-            self.date = $filter('date')(date, "MM-dd-yyyy");
         }
 
+        // Access the AngularJS scope
+        let scope = angular.element(document.querySelector('.container-calendar')).scope();
+
+        // Update the chosenDate on the scope
+        scope.$apply(function() {
+            scope.$ctrl.chosenDate = date;
+        });
+
+        console.log("Broadcasting date:", date);
         $rootScope.$broadcast('selectedDateChanged', date);
 
         renderCalendar();
-    });
+    });*/
+
+
+        //select dates
+        document.querySelector('.days').addEventListener('click', (event) => {
+            let target = event.target.dataset.target;
+            let date = new Date(); // Assuming you have a date object initialized already
+
+            if(target !== undefined) {
+                if (target.substring(0, 13) === "previous-page") {
+                    selectedDate = target.substring(14, 16).trim();
+                    date.setMonth(date.getMonth() - 1);
+                } else if (target.substring(0, 9) === "next-page") {
+                    selectedDate = target.substring(10, 12).trim();
+                    date.setMonth(date.getMonth() + 1);
+                } else {
+                    selectedDate = target;
+                    date.setDate(parseInt(target));
+                }
+                selectedTime = "";
+            }
+
+            // Access the AngularJS scope
+            let scope = angular.element(document.querySelector('.container-calendar')).scope();
+
+            // Update the chosenDate on the scope
+            scope.$apply(function() {
+                scope.$ctrl.chosenDate = date;
+            });
```

Comment:
hey mike, great work and thanks for commenting the logic but do you think you could give me a quick explanation of what's going on here?

Reply:
Ya sure, line 536 just gets the scope of the calendar (where the user selects a date)
Lines 539 to 541 just changes the chosenDate variable in the visits-controller.js file to the value chosen in the calendar.